ts/features/LandingScreenMultiBanner/utils/__tests__/bannerRenderableSelectors.test.ts

```diff
@@ -0,0 +1,87 @@
+import { isItwEnabledSelector } from "../../../../store/reducers/backendStatus";
+import { GlobalState } from "../../../../store/reducers/types";
+import { itwLifecycleIsValidSelector } from "../../../itwallet/lifecycle/store/selectors";
+import { isItwTrialActiveSelector } from "../../../trialSystem/store/reducers";
+import {
+  isItwDiscoveryBannerRenderableSelector,
+  renderabilitySelectorsFromBannerMap
+} from "../bannerRenderableSelectors";
+import { BannerMapById } from "../landingScreenBannerMap";
+
+jest.mock("../../../../store/reducers/backendStatus", () => ({
+  isItwEnabledSelector: jest.fn()
+}));
+jest.mock("../../../itwallet/lifecycle/store/selectors", () => ({
+  itwLifecycleIsValidSelector: jest.fn()
+}));
+jest.mock("../../../trialSystem/store/reducers", () => ({
+  isItwTrialActiveSelector: jest.fn()
+}));
+beforeEach(() => {
+  jest.resetAllMocks();
+  jest.clearAllMocks();
+});
+
+type JestMock = ReturnType<typeof jest.fn>;
+const mockValues = (
+  itwEnabledMock: boolean,
+  itwLifecycleValidMock: boolean,
+  trialActiveMock: boolean
+) => {
+  (isItwEnabledSelector as unknown as JestMock).mockReturnValue(itwEnabledMock);
+  (itwLifecycleIsValidSelector as unknown as JestMock).mockReturnValue(
+    itwLifecycleValidMock
+  );
+  (isItwTrialActiveSelector as unknown as JestMock).mockReturnValue(
+    trialActiveMock
+  );
+};
+
+describe("itwDiscoveryBannerSelector", () => {
+  for (const itwEnabled of [true, false]) {
+    for (const lifecycleValid of [true, false]) {
+      for (const trialActive of [true, false]) {
+        const expectedResult = !(!trialActive || lifecycleValid || !itwEnabled); // this mimicks the behaviour of the standalone banner's selector
+        it(`should return ${expectedResult} if isItwEnabled is ${itwEnabled}, trialActive is ${trialActive} and lifecycleValid is ${lifecycleValid}`, () => {
+          mockValues(itwEnabled, lifecycleValid, trialActive);
+          expect(
+            isItwDiscoveryBannerRenderableSelector({} as unknown as GlobalState)
+          ).toBe(expectedResult);
+        });
+      }
+    }
+  }
+});
+
+describe("renderabilitySelectorsFromBannerMap", () => {
+  it("should return a map of selectors, given  a dirty map", () => {
+    const bannerMap = {
+      ITW_DISCOVERY: {
+        isRenderableSelector: expect.any(Function),
```

Comment:
Why do not we set a specific function (defined with an external const with respect to the `bannerMap`) so the `toEqual` check later reports the reference equality?